query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/RestController.java

```diff
@@ -0,0 +1,65 @@
+package de.brandwatch.minianalytics.queryinput.controller;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import de.brandwatch.minianalytics.queryinput.postgres.model.Query;
+import de.brandwatch.minianalytics.queryinput.postgres.repository.QueryRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+
+@org.springframework.web.bind.annotation.RestController
+public class RestController {
+
+    Logger logger = LoggerFactory.getLogger(RestController.class);
+
+    @Autowired
+    QueryRepository queryRepository;
+
+    static final private Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    @RequestMapping(value = "/queries", method = RequestMethod.POST)
+    public ResponseEntity query(@RequestParam String query){
+        logger.info("API Call: query");
+        logger.info("Query: " + query);
+
+        //TODO parse Query into Lucene Query
+        try {
+            Query saveQuery = new Query(query);
+            queryRepository.save(saveQuery);
+
+            logger.info("Retrieved query: " + query);
+            return ResponseEntity.status(200).body(gson.toJson(query));
```

Comment:
It is good practice to return the persisted object in rest. You are just returning the object you have parsed the input string to. It might seem trivial, but can actually make a big difference.

The repository save method should actually return the object that it has successfully stored, so you can just use that.

Reply:
Done